### PR TITLE
[ios][linking] fix scheme result for universal links

### LINKED ISSUE #6609: [iOS] getInitialURL resolves to exps prefixed url on iOS standalone production app
### Environment

```
Expo version: 35.0.0
OS: iOS 13.3
appOwnership: standalone
expoReleaseChannel: production
device: iPhoneXs
```

### Steps to Reproduce

1. add Universal Link settings to your app.json following the [linking docs](https://docs.expo.io/versions/v36.0.0/workflow/linking/#universal-links-on-ios)

```json
    "ios": {
	  ...
      "associatedDomains": [
        "applinks:your.domain.com"
      ]
    },
```

2. hosting Apple App Site Association File at `your.domain.com`.

3. Click `your.domain.com` link from Browser.

### Expected Behavior

`Linking.getInitialURL()` resolves to `https://your.domain.com`.

### Actual Behavior

`Linking.getInitialURL()` resolves to `exps://your.domain.com`.

On Android production build and iOS dev build is work fine.
Only on iOS release build, `getInitialURL` reoslves to `exps://your.domain.com`.

### Reproducible Demo

No reproduciable demo, because I don't have any domains to associated with. :bowing_man: 

## Note

If someone face the issue with react-navigation and Universal Link, the workaround is adding RegExp of `exps://your.domain.com` to uriPrefix like below.

```js
const yourDomain = 'https:\\/\\/your\\.\\.domain\\.com\\/'
const expoWorkaroundDomain = 'exps:\\/\\/your\\.domain\\.com\\/'
const uriPrefix = new RegExp(
  `${ExpoLinking.makeUrl('/')}|${yourDomain}|${expoWorkaroundDomain}`
)
```

Thanks for Expo team maintaining awesome library :+1:.
Have a good holidays!

### PR DESCRIPTION
**Pull request by cruzach**
# Why

closes https://github.com/expo/expo/issues/6609

previously, behavior on standalone ios apps was for `Linking.getInitialURL()` called from a universal link to resolve to something like `exps://your.domain.com` instead of `https://your.domain.com`

# How
check if scheme is https, if it is then this was opened via a universal link. The one exception is if someone names the scheme of their app `https`, although I'm not sure how likely that is

# Test Plan

Tested on an expokit ios app, both locally and in testflight. Opening via a regular deeplink still results in `scheme://...`, and opening via a universal link results in `https://your.domain....`

### CODE REVIEW

**Review by esamelson — COMMENTED** [review]
does this break initial URLs in the expo client? wondering since i think (though not 100% sure) if we open a URL like `https://exp.host/@community/native-component-list` we actually expect the scheme to be replaced with `exps://` in that case.

**Comment by cruzach** [comment]
It doesn't look like it breaks it, I'm getting the same response in `getInitialUrl` on a build with this change as I am with the build from the App store (with the `exp://` scheme replacing `https://`). I think because, before just returning the `uri` the way it is, we check for both `isUniversalLink` *and* if the app is detached - https://github.com/expo/expo/blob/9edd2c4ae5bbf2f6f05e2f7d2560b0636365170c/ios/Exponent/Kernel/Services/EXKernelLinkingManager.m#L167-L170

**Review by esamelson — APPROVED** [review]
gotcha, makes sense! thanks!

**Comment by giautm** [comment]
Hi @cruzach , If i rebuild the app with SDKv36, is this fix has effect? Or I have to wait for a release?

**Comment by cruzach** [comment]
you'll have to wait for the next release, but if you're on SDK 36 you can use the workaround shared in the original issue

**Comment by LinusU** [comment]
Did this not make it in to SDK 37? Still seeing this bug there 🤔

**Comment by cruzach** [comment]
SDK 37 was released (March 31) before this PR landed, so it will be included in SDK 38

**Comment by LinusU** [comment]
Thanks for confirming!

**Pull request merged**
